plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/utils/UIUtils.java

```diff
@@ -485,7 +485,10 @@ public static ElementTreeSelectionDialog createFileDialog(String selectedFile,
             @Override
             public boolean select(Viewer viewer, Object parentElement, Object element) {
                 return element instanceof IContainer 
-                        || (element instanceof IFile && ((IFile)element).getFileExtension().equals(extension));
+                        || (element instanceof IFile 
+                        		&& extension != null 
```

Comment:
Javadoc to the method says that extension may be null. So the check before the comparison is correct, but what should be the logic for null? Could it be that if both extension and ((IFile)element).getFileExtension() are empty (null or empty string) that the method might return true? Or even if extension is null then filter has to accept all files?